docs/articles/documentation/guides/advanced-guides/test-https-features-and-http2-websites.md

```diff
@@ -23,21 +25,107 @@ If the tested page does not use HTTPS-specific features (like
+> Before you try HTTPS, launch testcafe with [`--hostname localhost` CLI parameter](../../reference/command-line-interface.md#--hostname-name) or [`hostname: localhost` configuration option](../../reference/configuration-file.md#hostname). Since all modern browsers thread localhost as a secure origin, some built-in browser features that require secure origin (Service workers, GeoAPI) will work. Third-party JS features like ApplePaySession may or may not work depending on implementation.
 
-The example below uses the PFX encoded private key and certificate chain to create an HTTPS server.
+### Use a Trusted Certificate
+
+To provide a certificate when you launch TestCafe with [CLI](../../reference/command-line-interface.md), use [`-ssl`](../../reference/command-line-interface.md#--ssl-options) option:
 
 ```sh
-testcafe --ssl pfx=path/to/file.pfx;rejectUnauthorized=true;...
+testcafe chrome  test.js --ssl pfx=path/to/trusted/certificate.pfx;rejectUnauthorized=true;
 ```
 
-When you use the programming interface, pass the HTTPS server options to the [createTestCafe](../../reference/testcafe-api/global/createtestcafe.md) method.
+### Use a Self-Signed Certificate
+
+[Generate a self-signed certificate with `openssl`](#generate-a-certificate-from-the-command-line) if you launch tests from the command line. In the API, you can use the [openssl-self-signed-certificate](https://www.npmjs.com/package/openssl-self-signed-certificate) module to [generate a certificate in code](#generate-a-certificate-in-your-code).
+
+#### Generate a Certificate From the Command Line
+
+In **bash**, you can use [OpenSSL](https://www.openssl.org/docs/man1.1.1/man1/openssl-req.html) to generate a certificate chain.
+
+1. Issue your own CA certificate and a CA-signed certificate for the web server:
+
+    ```sh
+    openssl genrsa -des3 -out myCA.key 2048
+    ```
+
+    You will be prompted for a non-empty passphrase. Take note of it.  
+2. Generate a root certificate:
+
+    ```sh
+    openssl req -x509 -new -nodes -key myCA.key -sha256 -days 1825 -out myCA.pem
+    ```
+
+    You will be prompted for additional information, press Enter to skip the questions.
+3. Create a CA-signed domain certificate and a `.csr` sign request:
+
+    ```sh
+    openssl genrsa -out testingdomain.key 2048
+    openssl req -new -key testingdomain.key -out testingdomain.csr
+    ```
+
+    You can press Enter to skip the questions, but you might want to set a recognizable CN (Common Name) so that you can easily find this certificate in a list of others later.  
```

Comment:
"but you might want to" and other phrases like "may or may not" make it sound like you are unsure. Be straightforward.

Reply:
Noted, many thanks